ui-particles-angular/projects/ui-particles-angular/src/lib/gio-license/gio-license.service.ts

```diff
@@ -0,0 +1,129 @@
+import { Observable, of } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { MatDialog } from '@angular/material/dialog';
+
+import { GioLicenseDialogComponent, GioLicenseDialogData } from './gio-license-dialog/gio-license-dialog.component';
+
+export type License = {
+  tier: string;
+  packs: Array<string>;
+  features: Array<string>;
+};
+
+export interface LicenseConfiguration {
+  resourceURL: string;
+  featureInfoData: Record<string, FeatureInfo>;
+  utmSource: string;
+  utmCampaign: string;
+}
+
+export interface UTM {
+  source: string;
+  medium: string;
+  campaign: string;
+}
+
+export interface LicenseOptions {
+  // We used feature name as utm_medium
+  feature?: string;
+  deployed?: boolean;
+  // Optional, useful if need a context in utm campaign (utm_content)
+  context?: string;
+}
+
+export interface FeatureInfo {
+  image?: string;
+  description?: string;
+  title?: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GioLicenseService {
+  private readonly loadLicense$: Observable<License>;
+
+  constructor(
+    private readonly http: HttpClient,
+    @Inject('LicenseConfiguration')
+    private readonly licenseConfiguration: LicenseConfiguration,
+    private readonly matDialog: MatDialog,
+  ) {
+    this.loadLicense$ = this.http.get<License>(licenseConfiguration.resourceURL).pipe(shareReplay(1));
+  }
+
+  public getLicense$(): Observable<License> {
+    return this.loadLicense$;
+  }
+
+  public isMissingFeature$(licenseOptions: LicenseOptions | undefined): Observable<boolean> {
+    if (licenseOptions?.deployed === false) {
+      return of(true);
+    }
+    if (licenseOptions?.feature == null || licenseOptions?.deployed === true) {
+      return of(false);
+    }
+    return this.getLicense$().pipe(
+      map(license => license == null || license.features.find(feat => feat === licenseOptions.feature) == null),
+    );
+  }
+
+  public getFeatureInfo(licenseOptions: LicenseOptions): FeatureInfo {
+    if (!licenseOptions.feature) {
+      throw new Error(`feature is undefined`);
+    }
+    const featureInfo = this.licenseConfiguration.featureInfoData[licenseOptions.feature];
+    if (!featureInfo) {
+      throw new Error(
+        `Unknown Feature value ${licenseOptions.feature}. Expected one of ${Object.keys(this.licenseConfiguration.featureInfoData)}`,
+      );
+    }
+    return featureInfo;
+  }
+
+  public getTrialURL(licenseOptions: LicenseOptions): string {
+    if (!licenseOptions.feature) {
+      throw new Error(`feature is undefined`);
+    }
+    let url = `https://gravitee.io/self-hosted-trial?utm_source=${this.licenseConfiguration.utmSource}&utm_medium=${licenseOptions.feature}&utm_campaign=${this.licenseConfiguration.utmCampaign}`;
+    if (licenseOptions.context) {
+      url += `&utm_content=${licenseOptions.context}`;
+    }
+    return url;
+  }
+
+  public openDialog(licenseOptions: LicenseOptions, event?: Event) {
```

Comment:
Suggestion: we could return the MatDialogRef to let the consumer calling this method do what he wants with the result ?